### PR TITLE
Trailing commas

### PR DESCRIPTION
**Pull request by eed3si9n**
Fixes https://github.com/tree-sitter/tree-sitter-scala/issues/157

Problem
-------
Trailing commas are not supported.

Solution
--------
This adds support for that.

### CODE REVIEW

File: script/smoke_test.sh [hunk 1]
```diff
@@ -3,8 +3,8 @@
 # This is an integration test to generally check the quality of parsing.
 
 SCALA_SCALA_LIBRARY_EXPECTED=100
-SCALA_SCALA_COMPILER_EXPECTED=67
-DOTTY_COMPILER_EXPECTED=66
+SCALA_SCALA_COMPILER_EXPECTED=66
+DOTTY_COMPILER_EXPECTED=65
```
Comment (keynmol) [inline script/smoke_test.sh:7]: Do you have any idea what regressed here?

Comment (eed3si9n) [reply script/smoke_test.sh:7]:
```
$ diff -U 0 main-reports/report-scala2-compiler.txt report-scala2-compiler.txt | rg '^[+-]\w+'
4:-scala/tools/nsc/ast/parser/TreeBuilder.scala                                 (ERROR [143, 60] - [143, 61])
7:+scala/tools/nsc/backend/jvm/analysis/BackendUtils.scala                      (ERROR [560, 6] - [627, 4])
10:+scala/tools/nsc/tasty/TreeUnpickler.scala                                   (ERROR [368, 10] - [378, 8])
13:+scala/tools/nsc/typechecker/Checkable.scala                                 (ERROR [320, 16] - [326, 13])
17:+scala/tools/nsc/typechecker/splain/SplainFormatting.scala                   (ERROR [360, 4] - [367, 2])
$ diff -U 0 main-reports/report-dotty-compiler.txt report-dotty-compiler.txt | rg '^[+-]\w+'
4:+src/dotty/tools/backend/sjs/JSCodeGen.scala                                                  (ERROR [805, 11] - [805, 12])
7:+src/dotty/tools/dotc/semanticdb/Scala3.scala                                                 (ERROR [163, 4] - [163, 36])
10:+src/dotty/tools/dotc/transform/ExtensionMethods.scala                                       (ERROR [117, 8] - [125, 6])
13:+src/dotty/tools/MainGenericCompiler.scala                                                   (ERROR [175, 21] - [175, 22])
16:+src/dotty/tools/repl/ParseResult.scala                                                      (ERROR [156, 44] - [156, 45])
19:+test-coursier/dotty/tools/coursier/CoursierScalaTests.scala                                 (ERROR [168, 175] - [168, 178])
23:+test/dotty/tools/dotc/SettingsTests.scala                                                   (ERROR [181, 98] - [181, 99])
```

If we take https://github.com/lampepfl/dotty/blob/3.2.1/compiler/src/dotty/tools/backend/sjs/JSCodeGen.scala as an example, the file actually contains a trailing comma: https://github.com/lampepfl/dotty/blob/3.2.1/compiler/src/dotty/tools/backend/sjs/JSCodeGen.scala#L104 so even if it was succeeding previously it was on some false basis. Now it's failing on line 805 due to leading infix (https://github.com/tree-sitter/tree-sitter-scala/issues/141).

Similar story with Scala 2 compiler. https://github.com/scala/scala/blob/v2.13.10/src/compiler/scala/tools/nsc/backend/jvm/analysis/BackendUtils.scala#L280-L288 has trailing commas.

Comment (ckipp01) [reply script/smoke_test.sh:7]: Ah, thanks for the example!

**Review by ckipp01 — APPROVED** [review]
Looks good. LGTM!

**Comment by ckipp01** [comment]
Interesting, I was curious what the % would do in Metals with this change since we use trailing commas quite a bit. It was the same, it dropped _just slightly_.

_Before_
`success percentage: 85.33%`

_After_
`success percentage: 85.11%`

**Pull request merged**